### PR TITLE
Revert Shared directory breaking changes from main

### PR DESCRIPTION
**Pull request by jatgarg**
Revert: https://github.com/microsoft/FluidFramework/pull/9087
Revert: https://github.com/microsoft/FluidFramework/pull/9451

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -45,15 +44,6 @@ Previously, the fluidErrorCode value (a pascaleCased term) was often used as the
 Now all error messages can be expected to be easily-read sentences,
 sometimes followed by a colon and an inner error message when applicable.
 
-
-### Doing operations not allowed on deleted sub directory
```
Comment (markfields) [inline BREAKING.md:49]: Will this move to the `Upcoming changes` section?  We should think about if/how to explain that this change was here but then was removed.  Maybe it's ok to just delete this for now though.

**Review by markfields — APPROVED** [review]
🎂  LGTM, and also see my comments about BREAKING.md.  Not sure how to handle this case there, maybe it's fine to just revert it, or maybe we should leave a note that in 0.58.2000 there was an accidental breaking change made (maybe don't even need to document it in depth).  @heliocliu what do you think?

File: BREAKING.md (hunk 1 above)
Comment (jatgarg) [reply BREAKING.md:49]: This change will be done in "next" branch right?

**Comment by heliocliu** [comment]
> 🎂 LGTM, and also see my comments about BREAKING.md. Not sure how to handle this case there, maybe it's fine to just revert it, or maybe we should leave a note that in 0.58.2000 there was an accidental breaking change made (maybe don't even need to document it in depth). @heliocliu what do you think?

i think it's fine to leave it out of the breaking.md for now.  deprecating the broken version on npmjs probably makes more sense for that

**Pull request merged**